host/tests/dsClasses/dsHdmiIn.py

```diff
@@ -0,0 +1,759 @@
+            return 0
+
+        portList = []
+        for i in range(0, numOfPorts):
+            portList.append(f'dsHDMI_IN_PORT_{i}')
+
+        return portList
+
+    def getVideoZoomModeList(self):
+        """
+        gets supported Zoom Mode as list.
+
+        Args:
+            None.
+
+        Returns:
+            A list of Zoom modes please refer dsvideozoommode enum class.
+        """
+
+        videoZoomModeList = []
+        for modeindex in hdmiInZoomMode:
+            videoZoomModeList.append(hdmiInZoomMode(modeindex).name)
+
+        return videoZoomModeList
+
+
+    def setHdmiInZoomMode(self, zoom_mode:str=0):
+        """
+        sets zoommode  on a HdmiIn port.
+
+        Args:
+            zoommode (str, optional): zoommode. Defaults to 0
+
+        Returns:
+            None.
+        """
+        promptWithAnswers = [
+            {
+                "query_type": "list",
+                "query": "Selected Zoom Mode",
+                "input": zoom_mode
+            }
+        ]
+
+        result = self.utMenu.select(self.testSuite, "Zoom Mode", promptWithAnswers)
+
+    def getEDIDVersionList(self):
+        """
+        gets supported EDID versions as list.
+
+        Args:
+            None.
+
+        Returns:
+            A list of EDID list please refer  enum class.
+        """
+
+        edidList = []
+        for edidindex in hdmiEdidVersion:
+            edidList.append(hdmiEdidVersion(edidindex).name)
+
+        return edidList
+
+    def setEdidVersion(self, port_type:str=0, edidversion:str=0):
+        """
+        sets edid version  on a particular HdmiIn port.
+
+        Args:
+            hdmiin_port (str): Defaults to 0
+            edidversion (str, optional): edidversion. Defaults to 0
+
+        Returns:
+            None.
+        """
+        promptWithAnswers = [
+             {
+                "query_type": "list",
+                "query": "List of supported ports:",
+                "input": str(port_type)
+             },
+             {
+                "query_type": "list",
+                "query": "Please select from availabe edid versions",
+                "input": edidversion
+             }
+        ]
+
+        result = self.utMenu.select(self.testSuite, "Set EdidVersion", promptWithAnswers)
+
+    def getEdidVersion(self, port_type:str=0):
+        """
+        Gets edidversion.
+
+        Args:
+            None.
+        Returns:
+            Edid version as string.
+        """
+        promptWithAnswers = [
+            {
+                "query_type": "list",
+                "query": "List of supported ports:",
+                "input": str(port_type)
+            }
+        ]
+
+        result = self.utMenu.select( self.testSuite, "Get EdidVersion", promptWithAnswers)
+
+        typeStatusPattern = r"Result dsGetEdidVersion IN:port:\[(\w+)\]:\[.*\] OUT:edidver:\[(HDMI_EDID_VER_\w+)\]:\[.*\],dsError_t:\[(dsERR_\w+)\]"
+        match = re.search(typeStatusPattern, result)
```

Comment:
assign edidversion to None, otherwise if there is no match, accessing edidversion will be error